homeassistant/components/brunt/__init__.py

```diff
@@ -1 +1,91 @@
 """The brunt component."""
+from __future__ import annotations
+
+import logging
+
+from aiohttp.client_exceptions import ClientResponseError, ServerDisconnectedError
+import async_timeout
+from brunt import BruntClientAsync, Thing
+
+from homeassistant.config_entries import SOURCE_IMPORT, ConfigEntry
+from homeassistant.const import CONF_PASSWORD, CONF_USERNAME
+from homeassistant.core import HomeAssistant
+from homeassistant.exceptions import ConfigEntryAuthFailed, ConfigEntryNotReady
+from homeassistant.helpers.aiohttp_client import async_get_clientsession
+from homeassistant.helpers.update_coordinator import DataUpdateCoordinator, UpdateFailed
+
+from .const import DOMAIN, REGULAR_INTERVAL
+
+PLATFORMS = ["cover"]
+
+_LOGGER = logging.getLogger(__name__)
+
+
+async def async_setup(hass: HomeAssistant, config: dict) -> bool:
+    """Component setup, run import config flow for each entry in config."""
+    if DOMAIN not in config:
+        return True
+    hass.async_create_task(
+        hass.config_entries.flow.async_init(
+            DOMAIN, context={"source": SOURCE_IMPORT}, data=config[DOMAIN]
+        )
+    )
+    return True
+
+
+async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
+    """Set up Brunt using config flow."""
+    hass.data.setdefault(DOMAIN, {})
+    bapi = BruntClientAsync(
+        username=entry.data[CONF_USERNAME],
+        password=entry.data[CONF_PASSWORD],
+        session=async_get_clientsession(hass),
+    )
+    try:
+        await bapi.async_login()
+    except ServerDisconnectedError as exc:
+        raise ConfigEntryNotReady("Brunt not ready to connect.") from exc
+    except ClientResponseError as exc:
+        raise ConfigEntryAuthFailed(
+            f"Brunt could not connect with username: {entry.data[CONF_USERNAME]}."
+        ) from exc
+    except Exception as exc:  # pylint: disable=broad-except
```

Comment:
We normally don't catch unknown exceptions in the integration besides in the config flow. Is there an error we expect here?